### ISSUE TITLE
Maven: Add option for generating code in different package per schema

### ISSUE THREAD

**Issue by Chessray** [joelittlejohn/jsonschema2pojo#492]
Hi,

The Maven plugin allows me to specify a package in which the generated code for all schemas will end up. Would it be possible to add an option that creates a subpackage for each schema based on the schema name?

**Comment by joelittlejohn** [comment]
This is possible already, you simply have to put each schema in a sub
directory. Each sub directory name will be appended to the package name for
any schema that appears there.

**Comment by Chessray** [comment]
That worked, thank you. :)

**Comment by luisespinal** [comment]
From what I just experienced, this works only if you use a single <sourceDirectory> property.

I had two source paths like this:

```
<sourcePaths>
  <sourcePath>${project.my.schemas}/foo</sourcePath>
  <sourcePath>${project.my.schemas}/bar</sourcePath>
</sourcePaths>

```
where each subdirectory (foo and bar) had their own schema file (named schema.json incidentally.)

In such a configuration, all POJOs were generated flatly under the package defined in <targetPackage>. 

So I had to change the plugin configuration to

`<sourceDirectory>${project.my.schemas}</sourceDirectory>
`
to get the desired effect.

Just FYI.
